hs_core/views/resource_folder_hierarchy.py

```diff
@@ -83,16 +82,22 @@ def data_store_structure(request):
             logical_file_type = ''
             logical_file_id = ''
             for f in ResourceFile.objects.filter(object_id=resource.id):
-                if name_with_rel_path == get_resource_file_name_and_extension(f)[0]:
+                if name_with_full_path == f.storage_path:
                     f_pk = f.pk
                     f_url = get_resource_file_url(f)
                     if resource.resource_type == "CompositeResource":
                         logical_file_type = f.logical_file_type_name
                         logical_file_id = f.logical_file.id
                     break
 
-            files.append({'name': fname, 'size': size, 'type': mtype, 'pk': f_pk, 'url': f_url,
-                          'logical_type': logical_file_type, 'logical_file_id': logical_file_id})
+            if f_pk != '':

```

Comment:
Done. Note that this only reports files that are synchronized with Django, so that dangling iRODS files "disappear" from broken resources. This means the user will only see files that can be operated upon meaningfully and will appear in manifests and metadata.